### DISCUSSION TITLE
Debouncing event handler in x-init

### DISCUSSION THREAD

**Discussion by skttl** [alpinejs/alpine#1044] [1. Help] [answered]
I'm trying to add scroll event listeners to x-refs (I don't want to add them in the DOM with @scroll.debounce, because I want it to be as portable as possible).

Codepen her: https://codepen.io/skttl/pen/vYXowBY?editors=1111

What I am trying to detect if the clientWidth of $refs.content is larger than $refs.wrapper. When the window resizes, this should be detected again.

In addition to that, I want to show an overlay (if content is overflowing), that should be removed when scrolled to the end.

I could do it by just slapping @resize.window.debounce on the root element, and @scroll.debounce on the wrapper element. But I want this component to be portable, by just adding x-data, x-init and x-refs, without worrying about attaching event listeners.

I tried adding lodash debounce to the eventlisteners, but the functions gets called for each event, and not debounced. Check the console.log for proof.

Can anyone help me get this right?

**Comment by skttl** [comment, marked as answer]
So turns out I am an idiot, and I just needed to move things around a bit :)

From
```js
window.addEventListener("resize", event => _.debounce(this.setFromResize(), 250));
this.$refs?.wrapper.addEventListener("scroll", event => _.debounce(this.setFromScroll(event)));
```

to
```js
window.addEventListener("resize", _.debounce(event => this.setFromResize(), 250));
this.$refs?.wrapper.addEventListener("scroll", _.debounce(event => this.setFromScroll(event)));
```

**Comment by KevinBatdorf** [comment]
> I could do it by just slapping @resize.window.debounce on the root element

You don't need to put the event listeners on the root element. Maybe that simplifies things?
